src/ukify/ukify.py

```diff
@@ -519,60 +511,34 @@ def call_systemd_measure(uki, opts):
 
     # PCR measurement
 
-    to_measure = []
-    tflist = []
-
-    # First, pick up the sections we shall measure now */
-    for s in uki.sections:
-        if not s.measure:
-            continue
-
-        if s.content is not None:
-            to_measure.append(f"--{s.name.removeprefix('.')}={s.content}")
-        else:
-            raise ValueError(f"Don't know how to measure section {s.name}");
-
-    # And now iterate through the base profile and measure what we haven't measured above
-    if opts.measure_base is not None:
-        pe = pefile.PE(opts.measure_base, fast_load=True)
-
-        # Find matching PE section in base image
-        for base_section in pe.sections:
-            name = pe_strip_section_name(base_section.Name)
+    # First, pick up either the base sections or the profile specific sections we shall measure now
+    to_measure = [s for s in uki.sections[profile_start:] if s.measure]
```

Comment:
```suggestion
    to_measure = {s.name: s for s in uki.sections[profile_start:] if s.measure}
```
This way we don't have to loop over all sections `to_measure` for every section.  It shouldn't make a difference because only the first section of a name gets added to `to_measure` anyway and on any version of Python recent enough to run ukify insertion order is preserved.

The test further down becomes
```diff
-if any(section.name == s.name for s in to_measure):
+if section.name in to_measure:
```
and the usage of `to_measure` could be
```diff
-            *(f"--{s.name.removeprefix('.')}={s.content}"
-              for s in to_measure),
+            *(f"--{s.name.removeprefix('.')}={s.content}"
+              for s in to_measure.values()),
```
or
```diff
-            *(f"--{s.name.removeprefix('.')}={s.content}"
-              for s in to_measure),
+            *(f"--{n.removeprefix('.')}={s.content}"
+              for n, s in to_measure.items()),
```

Reply:
Thinking about it, while this does reduce going through the sections from $O(n^2)$ to $O(n)$ `call_systemd_measure` is still called $O(\textrm{number of profiles})$ and the work is redone for every invocation. If people start adding ill-advised numbers of profiles, this could still be slower than necessary. Right now `call_systemd_measure` doesn't return anything, so the dict could be returned and reused.

Reply:
The limit for number of sections is ridiculously low, so we should be fine

Reply:
Ah, that I somehow missed, then it should be fine, though I think that the first change would still be a bit cleaner.

The limit is an implementation detail and not in the spec, is it? In that case, ukify should probably warn if more there are more profiles than the limit, though this can wait for another PR.